applications/nrf_desktop/src/modules/ble_scan.c

```diff
@@ -206,13 +207,19 @@ static int configure_filters(void)
 
 static void scan_start(void)
 {
+	size_t conn_count = count_conn();
+	size_t bond_count = count_bond();
 	int err;
 
-	if (count_conn() == CONFIG_BT_MAX_CONN) {
+	if (IS_ENABLED(CONFIG_DESKTOP_BLE_PEER_SCAN_REQUEST) &&
```

Comment:
In case we have no connected and one boned, we will look for new peers too, which is not intentional. I think we should change way filters are configured to handle this properly.
And maybe we could add `peers_only` as argument for this function.

Reply:
fixed

As for the other part of the comment - I originally done so but this proves to be problematic if we want to scan for several devices under one request. Now when click is done dongle will scan for one and then for the other device. Only when it fails to find the second one it will stop searching openly. We could confirm every device but it seems rather inconvenient.